Feature/scoreboard initial

davidrideout opened the pull request:
Scoreboard initial commit, work in progress.
Implements:
- login/logout/registration (manual registration untested)
- basic account page for editing zipcode and lobby group
- list of agenda items, agenda detail view, and a basic player for testimonies
- admin section for creating/editing agenda items

![image](https://user-images.githubusercontent.com/6044811/92821605-19a79d00-f380-11ea-99bf-e3b3cdc35276.png)
![image](https://user-images.githubusercontent.com/6044811/92821638-23310500-f380-11ea-834b-70d28566b14d.png)
![image](https://user-images.githubusercontent.com/6044811/92821662-27f5b900-f380-11ea-8f60-04e4f0c7d369.png)
![image](https://user-images.githubusercontent.com/6044811/92821676-2c21d680-f380-11ea-9425-46abbaa0a9cb.png)
![image](https://user-images.githubusercontent.com/6044811/92821702-32b04e00-f380-11ea-87bd-c31bf978a4f9.png)
![image](https://user-images.githubusercontent.com/6044811/92821732-39d75c00-f380-11ea-9e2c-ff4d53ad612a.png)
![image](https://user-images.githubusercontent.com/6044811/92821753-3f34a680-f380-11ea-9df8-22d85484cd96.png)

davidrideout committed:
Scoreboard Initial Commit

johnkmzhou commented:
Could you move the firebase files out of web and to the parent directory? Also, you don't need an env variable for the firebase api key, you can just commit it in code because it's meant to be public. Before you do it though, change the firebase rules to set `allow write: if false;` to prevent anyone from writing to it until we set up authorized admins.

davidrideout committed:
Move firebase files to root directory

davidrideout commented:
Files have been moved. I plan to commit firebase configs after the admin capability is done (toggling it for development is a hassle).

davidrideout committed:
Basic firestore rules setup

johnkmzhou commented on web/src/App.tsx:
```diff
@@ -0,0 +1,40 @@
+import React from 'react';
+import { Router } from 'react-router-dom';
+import { createBrowserHistory } from 'history';
```
history dependency is unnecessary. Use BrowserRouter instead.

davidrideout committed:
Minor refactoring

* Swapping moment for datefns

davidrideout committed:
Implement image uploads, storage permissions

* Refactoring in misc areas

davidrideout commented:
* Swapped moment for datefns
* Added firestore and storage permissions along with a basic claims capability (stored in /permissions/{uid})
* Added image upload for agenda items

davidrideout committed:
Add .env file

johnkmzhou reviewed (approved):
Looks like I forgot to submit the review I had last week with my comments but since it looks like you resolved most of the issues we've discussed, I'm going approve this one.
